### ISSUE TITLE
[Question] Why no hooks except for beforeSubmit even though they are in the docs and in @formio/js?

### ISSUE THREAD

**Issue by laurens-ritense** [formio/angular#1107]

**Comment by laurens-ritense** [comment]
I looked at the source code a bit and I don't even see them ever being implemented in the Angular version.

**Comment by laurens-ritense** [comment]
And my question is why? Shouldn't that functionality also be ported over from formiojs?

**Comment by ZenMasterJacob20011** [comment]
We should have these hooks added to formio in angular. We are happy to accept pull requests that add this behavior. If you need to add some behavior after the submit button is pressed we have some documentation on how to listen for events emitted by the form https://help.form.io/developers/form-development/form-renderer#form-events

**Comment by laurens-ritense** [comment]
The problem is that that list is not up to date because a lot of those form events are not implemented in the angular library like for example the nextPage event. So are these going to be implemented?

**Comment by ZenMasterJacob20011** [comment]
No, this issue currently is not a priority for are development cycle. However any pull requests that add the form event are welcome
